### PR TITLE
Add docstring for the torch.distributed.elastic.utils.distributed.get_free_port function

### LINKED ISSUE #127914: Document the torch.distributed.elastic.utils.distributed.get_free_port function
Document the `torch.distributed.elastic.utils.distributed.get_free_port` function located in [https://github.com/pytorch/pytorch/blob/main/torch/distributed/elastic/utils/distributed.py#L115](https://github.com/pytorch/pytorch/blob/main/torch/distributed/elastic/utils/distributed.py#L115).
Follow the [Docstring Guidelines](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines).

cc @brycebortree

### PR DESCRIPTION
**Pull request by afrittoli**
Fixes: #127914

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k @svekars @brycebortree

### CODE REVIEW

**Review by H-Huang — APPROVED** [review]
Thanks for adding it!

**Comment by afrittoli** [comment]
> Thanks for adding it!

Thanks for the review, I fixed the lint issue, sorry about that 🙏 
Is there a way I can trigger a re-run of the failed jobs only?

File: torch/distributed/elastic/utils/distributed.py [hunk 1]
```diff
@@ -113,6 +113,23 @@ def _check_full_rank(store, world_size, timeout):
...
+    Returns an unused port on localhost.
+
+    This function finds an unused port on localhost by opening to socket to bind
+    to a port and then closing it.
+
+    Returns:
+        int: an unused port on localhost
+
+    Example:
+        >>> get_free_port()
```
Comment (H-Huang) [inline torch/distributed/elastic/utils/distributed.py:126]: Please add `# xdoctest: +SKIP("Nondeterministic")` to fix the doctest errors. See https://github.com/pytorch/pytorch/blob/8892ddaaccf7f07d64e2e819d868c0e95bc53e74/torch/autograd/forward_ad.py#L82 as an example

Comment (afrittoli) [reply torch/distributed/elastic/utils/distributed.py:127]: Thank you for the suggestion, done now.

**Comment by afrittoli** [comment]
CI failures don't seem related to the PR:

```
manifest for nvidia/cuda:12.1-cudnn9-devel-ubuntu20.04 not found: manifest unknown: manifest unknown
```

**Pull request closed without merging**